vhd/packer/install-dependencies.sh

```diff
@@ -285,7 +285,7 @@ done
 
 VIRTUAL_KUBELET_VERSIONS="latest"
 for VIRTUAL_KUBELET_VERSION in ${VIRTUAL_KUBELET_VERSIONS}; do
-    CONTAINER_IMAGE="microsoft/virtual-kubelet:${VIRTUAL_KUBELET_VERSION}"
+    CONTAINER_IMAGE="mcr.microsoft.com/oss/virtual-kubelet/virtual-kubelet:${VIRTUAL_KUBELET_VERSION}"
```

Comment:
can we pin this? cc @cpuguy83

Reply:
It's always floated at `:latest`, but we could use `:0.9.1`. They point to the [same image](https://hub.docker.com/r/microsoft/virtual-kubelet/tags) and haven't been updated in a year.

Reply:
The hub image is not able to be updated anymore (per policy), which is why it was moved to ACR.
Honestly, I think the virtual-node add-on is using 'latest' as well 😞

Reply:
Looks like `microsoft/virtual-kubelet` registry is deprecated. published `v1.2.1.2` to MCR.

Reply:
I think pinning virtual-kubelet to a recent version is logically separate from this PR, so I'm going to merge this and we can follow up on #3810.